test/generate.test.ts

```diff
@@ -154,6 +154,73 @@ describe('generate', function () {
     });
   });
 
+  describe('scale-zero', () => {
+    it('should enumerate correct scale type when scale zero is used without bar mark or binning', () => {
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {
+              zero: true,
+              type: {values: [undefined, ScaleType.SQRT, ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC]}
+            },
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema);
+      assert.equal(answerSet.length, 2);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+      assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.SQRT);
+    });
+
+    it('should enumerate correct scale properties with mark bar', () => {
+      const specQ = {
+        mark: Mark.BAR,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {
+              zero: true,
+              type: {values: [undefined, ScaleType.SQRT, ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC]}
+            },
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema);
+      assert.equal(answerSet.length, 2);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+      assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.SQRT);
+    });
+
+    it('should enumerate correct scale properties with binned field and scale zero', () => {
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            bin: true,
+            channel: Channel.X,
+            scale: {
+              zero: true,
+              // type: {values: [undefined, ScaleType.SQRT, ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC]}
+            },
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema);
+      assert.equal(answerSet.length, 1); // shouldn't this be 0? why is undefined passing?
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
```

Comment:
@kanitw 
I do not understand why this is part of the answer set. I believe I wrote omitScaleZeroWithBinnedField correctly; the unit test is working as expected: if bin == true && zero == true, the constraint fails.

However in my generate test, this one case with undefined seems to be getting through. I'm not sure what I'm misunderstanding.